File: tests/c/hello.c
/* OneJoker card library <http://lcrocker.github.io/onejoker/cardlib>
 *
 * To the extent possibile under law, Lee Daniel Crocker has waived all
 * copyright and related or neighboring rights to this work.
 * <http://creativecommons.org/publicdomain/zero/1.0/>
 *
 * "Hello World" example.
 */

#include <stdio.h>
#include <stdlib.h>

#include "ojcardlib.h"

int main(int argc, char *argv[]) {
    int i, db[52], hb[5];
    char text[20];
    oj_cardlist_t deck, hand;

    printf("%d %d\n", sizeof(oj_cardlist_t), sizeof(oj_combiner_t));

    ojl_new(&deck, db, 52);
    ojl_fill(&deck, 52, OJD_STANDARD);
    ojl_shuffle(&deck);

    ojl_new(&hand, hb, 5);
    for (i = 0; i < 5; ++i) ojl_append(&hand, ojl_pop(&deck));

    ojl_text(&hand, text, sizeof(text));
    printf("%s\n", text);

    return 0;
}

File: tests/python/hello.py
#!/usr/bin/env python3
# OneJoker library <http://lcrocker.github.io/OneJoker>
#
# "Hello World" app included here to keep docs in sync with code.

import os, sys
sys.path.insert(1, os.path.dirname(__file__))
import ojcardlib as oj

if __name__ == "__main__":
    deck = oj.CardList(52)
    deck.fill()
    deck.shuffle()

    hand = oj.CardList(5)
    for i in range(5):
        hand.append(deck.pop())
    print(hand)
